### PR TITLE
EXPORT functionality

### PR DESCRIPTION
**Pull request by mih**
This should work. Please give it a go!

```
git annex initremote osftree type=external externaltype=osf encryption=none project=https://osf.io/<id> exporttree=yes
git annex export HEAD --to osftree
```

Fixes gh-56

### CODE REVIEW

**Comment by mjboos** [comment]
That's great!
Btw, it currently fails if no username/password is provided.
(did not mean to mark this as ready for review)

File: datalad_osf/remote.py [hunk 1]
```diff
@@ -181,11 +181,58 @@ def files(self):
...
+            fobj[0].write_to(fp)
+
+    def checkpresentexport(self, key, remote_file):
+        # return True if the file `remote_file` is present in the remote
+        # return False if not
+        # raise RemoteError if the presence of the file couldn't be determined,
+        # eg. in case of connection error
+        self.annex.info(key)
+        self.annex.info(remote_file)
+        return posixpath.sep + remote_file in set(f.path for f in self.files)
```
Comment (mjboos) [inline datalad_osf/remote.py:217]: This breaks for me, probably because the representation of self.files changed to a dictionary.

Comment (bpoldrack) [reply datalad_osf/remote.py:217]: Right. Should be `self.files.values()`

Comment (mih) [reply datalad_osf/remote.py:217]: FTR: All of this code is gone now.

**Comment by mih** [comment]
Feedback from chat with positive that it works (2x). Good enough for now.

**Pull request merged**